api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2479,6 +2484,68 @@ def test_anonymize_public_account(
         assert history.authorUser == legit_user
         assert history.user == user
 
+        response = authenticated_client.get(response.location)
+        assert "Les informations de l'utilisateur ont été anonymisées" in html_parser.extract_alert(response.data)
+
+    def test_anonymize_public_account_with_gdpr_extract(
+        self,
+        legit_user,
+        authenticated_client,
+    ):
+        user = users_factories.UserFactory(
+            gdprUserDataExtract=[
+                users_factories.GdprUserDataExtractBeneficiaryFactory(dateProcessed=datetime.datetime.today())
+            ]
+        )
+
+        with open(self.storage_folder / f"{user.gdprUserDataExtract[0].id}.zip", "wb"):
+            pass
+
+        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
+        assert response.status_code == 302
+
+        expected_url = url_for("backoffice_web.public_accounts.get_public_account", user_id=user.id, _external=True)
+        assert response.location == expected_url
+
+        assert "Anonymous" in user.firstName
+        assert user.roles == [users_models.UserRole.ANONYMIZED]
+        history = history_models.ActionHistory.query.one_or_none()
+        assert history.actionType == history_models.ActionType.USER_ANONYMIZED
+        assert history.authorUser == legit_user
+        assert history.user == user
```

Comment:
Pas besoin de tout retester, on peut se contenter de vérifier le role comme c'est fait dans d'autres tests sur le même sujet. par exemple dans `test_anonymize_public_account_when_user_has_no_deposit` quelques lignes lus bas
```suggestion
        assert user.roles == [users_models.UserRole.ANONYMIZED]

```